src/route/lwtunnel.rs

```diff
@@ -106,11 +126,156 @@ impl std::fmt::Display for RouteLwEnCapType {
+            Self::Hoplimit(_) => const { size_of::<u8>() },
+            Self::Tc(_) => const { size_of::<u8>() },
+            Self::Flags(_) => const { size_of::<u16>() },
+            Self::Other(_) => const { size_of::<DefaultNla>() },
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        match self {
+            Self::Unspecified => LWTUNNEL_IP6_UNSPEC,
+            Self::Id(_) => LWTUNNEL_IP6_ID,
+            Self::Destination(_) => LWTUNNEL_IP6_DST,
+            Self::Source(_) => LWTUNNEL_IP6_SRC,
+            Self::Hoplimit(_) => LWTUNNEL_IP6_HOPLIMIT,
+            Self::Tc(_) => LWTUNNEL_IP6_TC,
+            Self::Flags(_) => LWTUNNEL_IP6_FLAGS,
+            Self::Other(other) => other.kind(),
+        }
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        match self {
+            Self::Unspecified => {}
+            Self::Id(id) => NetworkEndian::write_u64(buffer, *id),
+            Self::Destination(ip) | Self::Source(ip) => {
+                buffer.copy_from_slice(&ip.octets());
+            }
+            Self::Hoplimit(value) | Self::Tc(value) => buffer[0] = *value,
+            Self::Flags(flags) => BigEndian::write_u16(buffer, flags.bits()),
+            Self::Other(other) => other.emit_value(buffer),
+        }
+    }
+}
+
+// should probably be in utils
+fn parse_u64_be(payload: &[u8]) -> Result<u64, DecodeError> {
+    if payload.len() != size_of::<u64>() {
+        return Err(format!("invalid u64: {payload:?}").into());
+    }
+    Ok(BigEndian::read_u64(payload))
+}
+
+impl<'a, T> Parseable<NlaBuffer<&'a T>> for RouteIp6Tunnel
+where
+    T: AsRef<[u8]> + ?Sized,
+{
+    fn parse(buf: &NlaBuffer<&'a T>) -> Result<Self, DecodeError> {
+        let payload = buf.value();
+        Ok(match buf.kind() {
+            LWTUNNEL_IP6_UNSPEC => Self::Unspecified,
+            LWTUNNEL_IP6_ID => Self::Id(
+                parse_u64_be(payload)
+                    .context("invalid LWTUNNEL_IP6_ID value")?,
+            ),
+            LWTUNNEL_IP6_DST => Self::Destination(
+                parse_ipv6_addr(payload)
+                    .context("invalid LWTUNNEL_IP6_DST value")?,
+            ),
+            LWTUNNEL_IP6_SRC => Self::Source(
+                parse_ipv6_addr(payload)
+                    .context("invalid LWTUNNEL_IP6_SRC value")?,
+            ),
+            LWTUNNEL_IP6_HOPLIMIT => Self::Hoplimit(
+                parse_u8(payload)
+                    .context("invalid LWTUNNEL_IP6_HOPLIMIT value")?,
+            ),
+            LWTUNNEL_IP6_TC => Self::Tc(
+                parse_u8(payload).context("invalid LWTUNNEL_IP6_TC value")?,
+            ),
+            LWTUNNEL_IP6_FLAGS => {
+                Self::Flags(RouteIp6TunnelFlags::from_bits_retain(
+                    parse_u16_be(payload)
+                        .context("invalid LWTUNNEL_IP6_FLAGS value")?,
+                ))
+            }
+            _ => {
+                return Err(DecodeError::from(
```

Comment:
Instead of return error on unknown `LWTUNNEL_IP6_XXX`, please use `RouteIp6Tunnel::Other` to store it. We can many existing code doing that, just copy them.